server/py/services/api/tests/unit/utils/singletons/test_k8s_utils.py

```diff
@@ -279,6 +279,27 @@ def test_store_secret(
         assert data == secrets_to_store
 
 
+
+def test_store_secrets_does_not_pass_labels_to_read_secret(k8s_helper):
+    """
+        Test ensures that labels param is not passed to read_secret when storing secrets.
+        The labels param during read_secret is intended for IG4 secrets only.
+    """
+    k8s_helper.read_secret = mock.MagicMock(
+        side_effect=k8s_dynamic_exceptions.NotFoundError(
+            k8s_client_rest.ApiException(status=404)
+        )
+    )
+    k8s_helper.store_secrets(
+        secret_name="my-secret",
+        secrets={"key1": "value1"},
+        namespace="default",
+    )
+    # Ensure 'labels' is not in the call arguments
+    for call in k8s_helper.read_secret.call_args_list:
+        assert "labels" not in call.kwargs
```

Comment:
can be simplified with a direct call argument assertion:
```suggestion
 k8s_helper.read_secret.assert_called_once_with(secret_name="my-secret", namespace="default")
```